source/standards/source-code/use-github.html.md.erb

```diff
@@ -52,12 +52,16 @@ Consider protecting the `.github/workflows` folder by using [a CODEOWNERS file](
 Consider creating a Workflow Template in the [alphagov workflow folder](https://github.com/alphagov/.github/tree/main/workflow-templates) if you need to share a similar workflow between many repositories.
 
 [Create your own local actions](https://docs.github.com/en/actions/creating-actions/about-actions) wherever possible. If using GitHub-owned actions, [pin to a specific version](https://docs.github.com/en/actions/learn-github-actions/finding-and-customizing-actions#using-release-management-for-your-custom-actions).
```

Comment:
```suggestion
[Create your own local actions](https://docs.github.com/en/actions/creating-actions/about-actions) wherever possible.

If using GitHub-owned actions, [pin to a specific version](https://docs.github.com/en/actions/learn-github-actions/finding-and-customizing-actions#using-release-management-for-your-custom-actions) and [configure Dependabot to keep your actions up to date](https://docs.github.com/en/code-security/dependabot/working-with-dependabot/keeping-your-actions-up-to-date-with-dependabot#example-dependabotyml-file-for-github-actions) by adding a comment on the same line with the tag the commitsha represents. For example:

`- uses: actions/checkout@01aecccf739ca6ff86c0539fbc67a7a5007bbc81 #v2.1.0`
```

Reply:
These are @andyloughran's changes from https://github.com/alphagov/gds-way/pull/874/files

Posting here so we can discuss what's worth pulling in?

Reply:
I quite like the code snippet, I was on the fence about putting one in, but decided to try and keep it all trimmer in a chunky section. What'd you think? Worth doing anyway?

Other thoughts:
- I think we should be clear we want version pinning on _any_ action? Potentially even ones we make ourselves no? So the distinction isn't between github actions / third party or internal?
- I _really_ want to talk about DependaBot here, it's soo useful in this context, but then erred, the dependency requirements in the [GDS way are more open](https://gds-way.digital.cabinet-office.gov.uk/standards/tracking-dependencies.html#update-dependencies-frequently). We reference DependaBot, but also allow for alternatives Snyk and some kinda python milarky. So wanted to outline it as a principle without calling out the impliemntation? I left more specific advice in the commit message though... perhaps that's too buried 🤔. I'd be interested in thoughts from likes of @timblair / @willp-bl  on that one, they've got a better birds-eye-view across GDS than me.

Reply:
Giving a specific example of how it works and noting that you don't have to use that specific tool/service should be ok. It will help explain what it is and you could link to the relevant guidance about dependencies

I can't think of a reason why we wouldn't have it on every action, even those we may make ourselves.

Reply:
Cool, I'll come back with an amendment here that nicks a bit more of Andy's code.

Reply:
Yeah, the only bit that is missing is that the Dependabot test fixture allows for many different types of version comment; whereas we're providing a single way.  There's no documentation outside of the fixture that I could find that explains what's supported or not.

Reply:
Cool, I'll allude to that